test/toplevel/test_virtualprocess.jl

```diff
@@ -1022,6 +1022,21 @@ end
 
         @test isempty(res.toplevel_error_reports)
     end
+
+    @testset "https://github.com/aviatesk/JET.jl/issues/280" begin
+        res = @analyze_toplevel begin
+            using Libdl
+            llvmpaths = filter(lib -> occursin(r"LLVM\b", basename(lib)), Libdl.dllist())
+            if length(llvmpaths) != 1
+                throw(ArgumentError("Found one or multiple LLVM libraries"))
+            end
+            libllvm = Libdl.dlopen(llvmpaths[1])
+            gethostcpufeatures = Libdl.dlsym(libllvm, :LLVMGetHostCPUFeatures)
+            ccall(gethostcpufeatures, Cstring, ())
+        end
```

Comment:
This top-level code is actually abstract-interpreted:
```julia
julia> res = @analyze_toplevel toplevel_logger=IOContext(stdout, :JET_LOGGER_LEVEL=>1) begin
           using Libdl
           llvmpaths = filter(lib -> occursin(r"LLVM\b", basename(lib)), Libdl.dllist())
           if length(llvmpaths) != 1
               throw(ArgumentError("Found one or multiple LLVM libraries"))
           end
           libllvm = Libdl.dlopen(llvmpaths[1])
           gethostcpufeatures = Libdl.dlsym(libllvm, :LLVMGetHostCPUFeatures)
           ccall(gethostcpufeatures, Cstring, ())
       end
[toplevel-debug] virtualized the context of Main (took 0.066 sec)
[toplevel-debug] concretization plan at REPL[2]:2:
1 t 1 ─     using Libdl
2 f └──     return nothing
[toplevel-debug] concretization plan at REPL[2]:3:
 1 t 1 ─       $(Expr(:thunk, CodeInfo(
    @ none within `top-level scope`
1 ─      global var"#1#2"
│        const var"#1#2"
│   %3 = Core._structtype(Main.##JETVirtualModule#278, Symbol("#1#2"), Core.svec(), Core.svec(), Core.svec(), false, 0)
│        var"#1#2" = %3
│        Core._setsuper!(var"#1#2", Core.Function)
│        Core._typebody!(var"#1#2", Core.svec())
└──      return nothing
)))
 2 t │   %2  = Core.svec(var"#1#2", Core.Any)
 3 t │   %3  = Core.svec()
 4 t │   %4  = Core.svec(%2, %3, $(QuoteNode(:(#= REPL[2]:3 =#))))
 5 t │         $(Expr(:method, false, :(%4), CodeInfo(
    @ REPL[2]:3 within `none`
1 ─ %1 = basename(lib)
│   %2 = occursin(r"LLVM\b", %1)
└──      return %2
)))
 6 f │         _1 = %new(var"#1#2")
 7 f │   %7  = _1
 8 f │   %8  = Base.getproperty(Libdl, :dllist)
 9 f │   %9  = (%8)()
10 f │   %10 = filter(%7, %9)
11 f │         llvmpaths = %10
12 f └──       return %10
[toplevel-debug] concretization plan at REPL[2]:4:
1 f 1 ─ %1 = length(llvmpaths)
2 f │   %2 = %1 != 1
3 f └──      goto #3 if not %2
4 f 2 ─ %4 = ArgumentError("Found one or multiple LLVM libraries")
5 f │   %5 = throw(%4)
6 f └──      return %5
7 f 3 ─      return nothing
[toplevel-debug] concretization plan at REPL[2]:7:
1 f 1 ─ %1 = Base.getproperty(Libdl, :dlopen)
2 f │   %2 = Base.getindex(llvmpaths, 1)
3 f │   %3 = (%1)(%2)
4 f │        libllvm = %3
5 f └──      return %3
[toplevel-debug] concretization plan at REPL[2]:8:
1 f 1 ─ %1 = Base.getproperty(Libdl, :dlsym)
2 f │   %2 = (%1)(libllvm, :LLVMGetHostCPUFeatures)
3 f │        gethostcpufeatures = %2
4 f └──      return %2
[toplevel-debug] concretization plan at REPL[2]:9:
1 f 1 ─ %1 = $(Expr(:foreigncall, :gethostcpufeatures, :Cstring, :(Core.svec()), 0, :(:ccall)))
2 f └──      return %1
```
(and thus this testcase passes without this PR)

Reply:
After a closer look, it seems like the root problem here is not the missing support for `:foreigncall` expressions, but inappropriate statement selections.
I could successfully analyze `ScanByte` by this configuration: `report_package("ScanByte", toplevel_logger=IOContext(stdout, :JET_LOGGER_LEVEL=>1), concretization_patterns=[:(const x_ = y_), :(let; x_; end)])`

Reply:
Sorry, needed to wrap the code into a (let) block.

Reply:
Yeah, well, my point was that the latest master will pass the test case even when it's wrapped into `let` block -- I think it implies `:foreigncall` expression tweak is _not_ necessary for concretization by JuliaInterpreter.